src/app/cli/src/commands/delete_command.rs

```diff
@@ -58,10 +58,16 @@ impl DeleteCommand {
 #[async_trait::async_trait(?Send)]
 impl Command for DeleteCommand {
     async fn validate_args(&self) -> Result<(), CLIError> {
-        if self.dataset_ref_patterns.is_empty() && !self.all {
-            Err(CLIError::usage_error("Specify a dataset or use --all flag"))
-        } else {
-            Ok(())
+        match (self.dataset_ref_patterns.len(), self.all) {
+            (0, true) => Ok(()),
+            (len, false) if len > 0 => Ok(()),
+            (0, false) => Err(CLIError::usage_error("Specify dataset(s) or pass --all")),
+            (len, true) if len > 0 => Err(CLIError::usage_error(
+                "You can either specify dataset(s) or pass --all",
+            )),
+            _ => Err(CLIError::usage_error(
+                "Invalid combination of arguments".to_owned(),
+            )),
```

Comment:
Legend:
```rust
/*1*/  (0, true) => Ok(()),
/*2*/  (len, false) if len > 0 => Ok(()),
/*3*/  (0, false) => Err(CLIError::usage_error("Specify dataset(s) or pass --all")),
/*4*/  (len, true) if len > 0 => Err(CLIError::usage_error(
                "You can either specify dataset(s) or pass --all",
            )),
/*5*/  _ => Err(CLIError::usage_error(
                "Invalid combination of arguments".to_owned(),
            )),
```

---

a) 
```rust
/*1*/  (0, true) => Ok(()),
/*4*/  (len, true) if len > 0 => Err(CLIError::usage_error(
                "You can either specify dataset(s) or pass --all",
            )),
```
We don't need an additional predicate (`if len > 0`) for `/*4*/` since the zero length is handled by branch `/*1*/`

b) Same for pair `/*2*/` and `/*3*/`
c) It looks like branch `/*5*/` is not needed since it can't be called

Reply:
(Optional) d) Please sort the branches, e.g. by pairs, or in ascending order. We have four options here, roughly:
```
0 0
0 1
1 0
1 1
```

In my opinion, it might simplify the readability a little bit

Reply:
Not agree to remove guard, as matches order change may affect logic which I would like to avoid. But agree that it looks like it needs improvement. I changed to matching against slice.
Order was changed as requested.